src/jstests/formio/components/file.spec.js

```diff
@@ -66,4 +66,30 @@ describe('Multiple File Component', () => {
       })
       .catch(done);
   });
+
+  // GH-4222
+  test('Uploading 1 file then 2 files gives an error', done => {
+    let formJSON = _.cloneDeep(maxNFilesForm);
+
+    const element = document.createElement('div');
+
+    Formio.createForm(element, formJSON)
+      .then(form => {
+        form.setPristine(false);
+        const component = form.getComponent('multipleFiles');
+        component.upload([{name: 'File 1', size: '50'}]);
+
+        component.upload([
+          {name: 'File 2', size: '50'},
+          {name: 'File 3', size: '50'},
+        ]);
```

Comment:
internally this is called `redraw()`, which returns a promise but nowhere do they actually await this promise. So I'm expecting some race conditions here - what happens if you put some `sleep` calls before and after the `upload` calls?

Reply:
I can make it work by pushing an initial file in `component.dataValue`